src/main/java/io/supertokens/config/CoreConfig.java

```diff
@@ -92,9 +92,19 @@
     private String base_path = "";
 
     public String getBasePath() {
-        if (base_path == null || base_path.equals("/")) {
+        String base_path = this.base_path; // Don't modify the original value from the config
+        if (base_path == null || base_path.equals("/") || base_path.isEmpty()) {
             return "";
         }
+        if (base_path.contains(" ")) {
+            throw new QuitProgramException("Invalid characters in base_path config");
+        }
+        if (!base_path.startsWith("/") { // Add leading '/'
+            base_path = "/" + base_path;
+        }
+        if (base_path.endsWith("/") { // Remove trailing '/'
+            base_path = base_path.substring(0, base_path.length() - 1);
+        }
```

Comment:
Actually, I had forgotten that these checks were happening in the validation function below. So the modifications here are unnecessary. You can revert this back to how it used to be.